### DISCUSSION TITLE
`ClearHistoryAndReset` is misleading and can harm your terminal experience.

### DISCUSSION THREAD

**Discussion by christianparpart** [contour-terminal/contour#921] [Development]
Not sure if anyone has ever tried to use that action intentionally in a situation where you would never want to do that, e.g. while being in an active and health alternate-screen app, such as vim / htop / mc / etc...

It'll actually severely hurt your App and for the `mc` case I couldn't cleanly exit anymore either. Such an action should be really taken with care.

OTOH, this action should really (!) be renamed from `ClearHistoryAndReset` to `HardReset`, because that's what it is doing. The history-clearance is implicitly part of it.

Renaming this action (if we want to keep it) to `HardReset` should hopefully raise the awareness of the severity by the user that wishes to trigger it.

Am I too sensitive here?

**Comment by data-man** [comment]
#263
Maybe change it to `ClearHistory` with `Reset` parameter?

**Reply by christianparpart** [reply]
@data-man what's your intent if you trigger this action? Why do you want to reset in the first place? What do you hope here for?

Adding `reset` as a parameter to `ClearHistory`-action would be wrong because reset implies clearing the history, because that's part of the reset.
I'd like to understand what you want to accomplish here. Maybe we should fully redesign this action - generally speaking.

Maybe we should check `konsole` to see what they do as part of the Reset action. The VT standard provides a soft reset (DECSTR) and a hard reset (RIS). I kinda doubt RIS is meant, due to its implications and fact that it'll actually break running applications (as our current application can do while being triggered while in alt-mode).

**Reply by data-man** [reply]
During the compilation of large projects (e.g. kernel) or by using wget, curl, etc, if I am not interested in some lines, I press Ctrl-L.
And later I analyze the remaining lines.
In Konsole nothing happens when I press `Ctrl-L` in active mc, htop, etc.
